lib/charms/acme_client_operator/v0/acme_client.py

```diff
@@ -111,6 +127,26 @@ def validate_generic_acme_config(self) -> bool:
             return False
         return True
 
+    @abstractmethod
+    def _on_config_changed(self, event: EventBase) -> None:
```

Comment:
I guess we could have observed on config changed here and use callable to retrieve configuration values from the charm. But that would have departed from the existing structure of this base charm.

Reply:
Exactly my thinking. It would still be possible to do it that way, but we would need a new major version and a lot more work on children charms.

Doing that right away when we do not have a lot of children charm might be faster, but we could also push this redesign when we have a more substantial change to make.

Reply:
Either way, to fix the bug without a major version, this is the only design that works, so we should probably still get this in first.